cmd/pj-rehearse/main_test.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	clienttesting "k8s.io/client-go/testing"
+	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/client/clientset/versioned/fake"
+	prowconfig "k8s.io/test-infra/prow/config"
+	prowgithub "k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/kube"
+	pjdwapi "k8s.io/test-infra/prow/pod-utils/downwardapi"
+
+	"github.com/openshift/ci-operator-prowgen/pkg/diffs"
+	"github.com/openshift/ci-operator-prowgen/pkg/rehearse"
+)
+
+var (
+	candidatePath          = "../../test/pj-rehearse/candidate"
+	masterPath             = "../../test/pj-rehearse/master"
+	candidateConfigPath    = filepath.Join(candidatePath, diffs.ConfigInRepoPath)
+	candidateJobConfigPath = filepath.Join(candidatePath, diffs.JobConfigInRepoPath)
+	masterConfigPath       = filepath.Join(masterPath, diffs.ConfigInRepoPath)
+	masterJobConfigPath    = filepath.Join(masterPath, diffs.JobConfigInRepoPath)
+	logger                 = logrus.WithFields(logrus.Fields{prowgithub.OrgLogField: jobSpec.Refs.Org, prowgithub.RepoLogField: jobSpec.Refs.Repo})
+	loggers                = rehearse.Loggers{Job: logger, Debug: logger}
+	jobSpec                = pjdwapi.JobSpec{
+		Type:      kube.PresubmitJob,
+		Job:       "pull-pj-rehearse-demo",
+		BuildID:   "0",
+		ProwJobID: "prowjob",
+		Refs: &kube.Refs{
+			Org:     "super",
+			Repo:    "duper",
+			BaseRef: "base-ref",
+			BaseSHA: "base-sha",
+			Pulls: []v1.Pull{{
+				Number: 2596,
+				Author: "dptp-team",
+				SHA:    "a4dc6278a189aa66ae9c2a0f11834499f104fcd4",
+			}},
+		},
+	}
+)
+
+func TestIntergration(t *testing.T) {
+	prowConfig, err := prowconfig.Load(masterConfigPath, masterJobConfigPath)
+	if err != nil {
+		t.Fatalf("failed to load Prow config: %v", err)
+	}
+	prowPRConfig, err := prowconfig.Load(candidateConfigPath, candidateJobConfigPath)
+	if err != nil {
+		t.Fatalf("failed to load Prow config: %v", err)
+	}
+
+	fakecs := fake.NewSimpleClientset()
+	fakeclient := fakecs.ProwV1().ProwJobs("test-namespace")
+
+	changedPresubmits := diffs.GetChangedPresubmits(prowConfig, prowPRConfig, logger)
+
+	watcher, err := fakeclient.Watch(metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Failed to setup watch: %v", err)
+	}
+
+	fakecs.Fake.PrependWatchReactor("prowjobs", func(clienttesting.Action) (bool, watch.Interface, error) {
+		watcher.Stop()
+		size := 0
+		for _, jobs := range changedPresubmits {
+			size += len(jobs)
+		}
+		ret := watch.NewFakeWithChanSize(size, true)
+		for event := range watcher.ResultChan() {
+			pj := event.Object.(*v1.ProwJob).DeepCopy()
+			pj.Status.State = v1.SuccessState
+			ret.Modify(pj)
+		}
+		return true, ret, nil
+	})
+
+	executor := rehearse.NewExecutor(changedPresubmits, jobSpec.Refs.Pulls[0].Number, candidatePath, jobSpec.Refs, true, loggers, fakeclient)
```

Comment:
I think all that's left is checking that we created the correct prowjobs, right?